chapters/naming-things.qmd

```diff
@@ -0,0 +1,733 @@
+2021-06-11_NEOFFI_emotional-stroop_msc-thesis_004.csv
+```
+
+## {{< fa brands apple >}} macOS
+
+Put image of macOS screenshot here
+
+## {{< fa brands r-project >}}
+
+```{zsh filename="Code"}
+list.files(pattern = "emotional") |> head
+```
+
+```{r filename="Output"}
+#| code-copy: false
+[1] "2021-06-04_NEOFFI_emotional-stroop_msc-thesis_001.csv"
+[2] "2021-06-05_NEOFFI_emotional-stroop_msc-thesis_002.csv"
+[3] "2021-06-07_NEOFFI_emotional-stroop_msc-thesis_003.csv"
+[4] "2021-06-11_NEOFFI_emotional-stroop_msc-thesis_004.csv"
+```
+
+:::
+
+## Biology
+
+::: {.panel-tabset}
+
+## {{< fa terminal >}} Bash
+
+```{zsh filename="Code"}
+ls *Plasmid*
+```
+
+```{zsh filename="Output"}
+#| code-copy: false
+2013-06-26_BRAFWTNEGASSAY_Plasmid-Cellline-100-1MutantFraction_H01.csv
+2013-06-26_BRAFWTNEGASSAY_Plasmid-Cellline-100-1MutantFraction_H02.csv
+2013-06-26_BRAFWTNEGASSAY_Plasmid-Cellline-100-1MutantFraction_H03.csv
+```
+
+## {{< fa brands apple >}} macOS
+
+Put image of macOS screenshot here
+
+## {{< fa brands r-project >}}
+
+```{zsh filename="Code"}
+list.files(pattern = "Plasmid") |> head
+```
+
+```{r filename="Output"}
+#| code-copy: false
+[1] "2013-06-26_BRAFWTNEGASSAY_Plasmid-Cellline-100-1MutantFraction_A01.csv"
+[2] "2013-06-26_BRAFWTNEGASSAY_Plasmid-Cellline-100-1MutantFraction_A02.csv"
+[3] "2013-06-26_BRAFWTNEGASSAY_Plasmid-Cellline-100-1MutantFraction_A03.csv"
+[4] "2013-06-26_BRAFWTNEGASSAY_Plasmid-Cellline-100-1MutantFraction_B01.csv"
+[5] "2013-06-26_BRAFWTNEGASSAY_Plasmid-Cellline-100-1MutantFraction_B02.csv"
+[6] "2013-06-26_BRAFWTNEGASSAY_Plasmid-Cellline-100-1MutantFraction_B03.csv"
+```
+
+:::
+
+:::
+
+### Recover meta data from file names
+
+Deliberate use of underscores (`_`) and hyphens (`-`) allows us to recover metadata from file names.
+
+- Use underscores (`_`) to delimit units of metadata
+- Use hyphens (`-`) to delimit words to improve human readability
+
+::: panel-tabset
+
+## Psychology
+
+::: {.panel-tabset}
+
+## {{< fa brands r-project >}}
+
+```{r filename="Code"}
+flist <- list.files(pattern = "emotional") %>% head
+stringr::str_split_fixed(flist, "[_\\.]", 6)
+```
+
+```{r filename="Output"}
+#| code-copy: false
+# date survey paradigm project participant
+      [ ,1]        [ ,2]    [ ,3]              [ ,4]        [ ,5] [ ,6]  
+[1, ] "2021-06-04" "NEOFFI" "emotional-stroop" "msc-thesis" "001" "csv"
+[2, ] "2021-06-05" "NEOFFI" "emotional-stroop" "msc-thesis" "002" "csv"
+[3, ] "2021-06-07" "NEOFFI" "emotional-stroop" "msc-thesis" "003" "csv"
+[4, ] "2021-06-11" "NEOFFI" "emotional-stroop" "msc-thesis" "004" "csv"
+```
+
+:::
+
+## Biology
+
+::: {.panel-tabset}
+
+## {{< fa brands r-project >}}
+
+```{r filename="Code"}
+flist <- list.files(pattern = "Plasmid") %>% head
+stringr::str_split_fixed(flist, "[_\\.]", 5)
+```
+
+```{r filename="Output"}
+#| code-copy: false
+# date assay sampleset well
+     [ ,1]        [ ,2]            [ ,3]                                  [ ,4] [ ,5]
```

Comment:
What do you think of adding these column indices @lnnrtwttkhn?

Reply:
Fine, I think!